nibabel/openers.py

```diff
@@ -145,3 +145,13 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close_if_mine()
+
+    @classmethod
+    def register_extension(opener_klass, ext, func):
+        """Decorator"""
+        def decorate(klass):
+            assert ext not in opener_klass.compress_ext_map, \
+                "Cannot redefine extension-function mappings."
+            opener_klass.compress_ext_map[ext] = func
```

Comment:
Forgive my anxiety here, but this decorator, though clever, makes me anxious.  It's unexpected to me that a class decorator should alter the global state - in this case of attributes in another class.  It makes it harder to reason about `Opener`.  For example, if someone imports `Opener` from outside nibabel, they will have different file extensions depending on whether freesurfer stuff got imported or no.

Can you see any way round that?

Can you see any way round this?